### DISCUSSION TITLE
Unable to test new changes without a local commit

### DISCUSSION THREAD

**Discussion by sandeep-lal** [cypress-io/cypress#27374] [Questions and Help]
Hi I seem to be having a lot of problems running my code without committing. I've managed to get the spec files running fine. The problem I have is I have a 2nd cypress.config-prod.js file which has prod credentials. I have to manually run these to ensure they are not stored in our remote repo but cypress does not seem to pick up the changes unless I commit.

**Comment by portyu9** [comment]
Hi @sandeep-lal 

A Git commit should not be required for Cypress to see local changes — Cypress runs against the files in your working tree, not only the last committed revision.

One thing I would check first is how `cypress.config-prod.js` is being selected. Cypress will not automatically use a second config file just because it exists. If that is the configuration you want for the run, explicitly launch Cypress with it:

```bash
npx cypress open --config-file cypress.config-prod.js
```

or:

```bash
npx cypress run --config-file cypress.config-prod.js
```

If you are already doing that, try completely closing and reopening Cypress after changing the config and confirm under **Settings → Project Settings / resolved configuration** that the values are actually coming from `cypress.config-prod.js`.

I would also avoid storing production credentials directly in that file, even if the file is ignored by Git. Keep the configuration structure in source control and inject the credentials through environment variables or your CI/CD secret store instead.

For example:

```bash
CYPRESS_PROD_USERNAME="..." \
CYPRESS_PROD_PASSWORD="..." \
npx cypress run --config-file cypress.config-prod.js
```

That keeps the secrets outside the repository while still allowing the production-specific configuration to be selected explicitly.

If committing the file genuinely changes the behavior, I would also inspect any wrapper script, Docker command, npm script, or other tooling used to start Cypress. Something there may be rebuilding/copying files from Git rather than running directly from the current working tree, because Cypress itself should not require a commit before local configuration changes become visible.
